File: scripts/plot_correlations.py
import numpy as np
import corner

m1pct = np.loadtxt('../paper/figures/m1pct_BPLG__including_230529_3.0.txt')
mu = np.loadtxt('../paper/figures/mu_BPLG__including_230529_3.0.txt')
sigma = np.loadtxt('../paper/figures/sigma_BPLG__including_230529_3.0.txt')

data = np.array([m1pct, mu, sigma]).T

fig = corner.corner(data, labels = [r'$m_{1\%}$', r'$m_{peak}$', r'$\sigma_{peak}$'], plot_datapoints = False, smooth = 1.2)#, levels = [1])
fig.savefig('../paper/figures/correlations.pdf')

File: scripts/plot_correlations.jl
using Pkg
Pkg.activate(joinpath(@__DIR__, ".."))
Pkg.add(PackageSpec(name="StatsPlots", version="0.14.33"))

using DelimitedFiles, DataFrames, StatsPlots, LaTeXStrings, Plots

# Load from text files (assumes one column of numbers per file)
m1pct = readdlm(joinpath(@__DIR__, "..", "paper", "figures","m1pct_BPLG__including_230529_3.0.txt"))[:]
mu = readdlm(joinpath(@__DIR__, "..", "paper", "figures","mu_BPLG__including_230529_3.0.txt"))[:]
sigma = readdlm(joinpath(@__DIR__, "..", "paper", "figures","sigma_BPLG__including_230529_3.0.txt"))[:]

# Assemble into DataFrame with LaTeX-style labels
df = DataFrame(L"m_{1\%}" => m1pct, L"m_{peak}" => mu, L"\sigma" => sigma)

# Generate corner plot
p = @df df StatsPlots.pairplot([L"m_{1\%}", L"m_{peak}", L"\sigma"], label = "", grid = false)
savefig(p, joinpath(@__DIR__, "..", "paper", "figures", "correlations_3.pdf"))
